src/huggingface_hub/inference/_providers/hf_inference.py

```diff
@@ -24,15 +24,19 @@ def _prepare_api_key(self, api_key: Optional[str]) -> str:
         return api_key or get_token()  # type: ignore[return-value]
 
     def _prepare_mapped_model(self, model: Optional[str]) -> str:
-        if model is not None:
+        if model is not None and model.startswith(("http://", "https://")):
             return model
-        model = _fetch_recommended_models().get(self.task)
         if model is None:
-            raise ValueError(
-                f"Task {self.task} has no recommended model for HF Inference. Please specify a model"
-                " explicitly. Visit https://huggingface.co/tasks for more info."
-            )
-        return model
+            model_id = _fetch_recommended_models().get(self.task)
+            if model_id is None:
+                raise ValueError(
+                    f"Task {self.task} has no recommended model for HF Inference. Please specify a model"
+                    " explicitly. Visit https://huggingface.co/tasks for more info."
+                )
+        else:
+            model_id = model
```

Comment:
I feel like it's slightly more readable like this

```suggestion
        model_id = model if model is not None else _fetch_recommended_models().get(self.task)
        if model_id is None:
            raise ValueError(
                f"Task {self.task} has no recommended model for HF Inference. Please specify a model"
                " explicitly. Visit https://huggingface.co/tasks for more info."
            )
```